modules/nlu/src/backend/training-pipeline.ts

```diff
@@ -542,13 +546,15 @@ export const Trainer: Trainer = async (input: TrainInput, tools: Tools): Promise
       list_entities: output.list_entities,
       oos_model,
       tfidf: output.tfIdf,
-      ctx_model,
       intent_model_by_ctx,
       slots_model,
       vocabVectors: output.vocabVectors,
       exact_match_index
       // kmeans: {} add this when mlKmeans supports loading from serialized data,
     }
+    if (_.isString(ctx_model)) {
```

Comment:
are you sure this is necessary ? 

in the case where the context model is undefined, it'll add ctx_model:undefined in artefact then won't be serialized as

```ts
JSON.stringify({ctx_model:undefined}) === "{}"
```

or maybe I don't understand what you're trying to do here ?